libs/sdk-ui-dashboard/src/dashboard/eventEmitter/rootEventEmitter.ts

```diff
@@ -0,0 +1,45 @@
+// (C) 2021 GoodData Corporation
+import { actionChannel, take } from "redux-saga/effects";
+import { DashboardEvents } from "../../events/dashboard";
+import { DashboardEventHandler } from "../../events/eventHandler";
+
+export type EventEmitter = {
+    registerHandler: (handler: DashboardEventHandler) => void;
+    // TODO: type this
+    eventEmitterSaga: any;
+};
+
+/**
+ * Creates root event emitter that will be responsible for emitting events to all registered handlers.
+ *
+ * The emitter is realized using a saga. This saga has its own dedicated channel through which it receives
+ * the events to emit - the event bus. Upon emitter start, it creates the channel and sets it into the
+ * `eventEmitterContext` field of the saga context - this way other sagas can get hold of it.
+ *
+ * @param initialHandlers - event handlers to register at the time of creation
+ */
+export function createRootEventEmitter(initialHandlers: DashboardEventHandler[] = []): EventEmitter {
+    const eventHandlers = initialHandlers;
+
+    return {
+        registerHandler: (handler: DashboardEventHandler) => {
```

Comment:
Do we need also a way to "unregister"?

Reply:
yep that's highly likely. i only added register for now to verify that the 'pipeline' works.

i suspect there will be more regarding the event subscriptions and their convenience.